### DISCUSSION TITLE
Transient simulations, non-dimensionalization, and equivalent circuit networks

### DISCUSSION THREAD

**Discussion by TomTranter** [PMEAL/OpenPNM#1879] [Development]
We are already aware that the transient simulations are physically suspicious regarding the use of our diffusion analogies for ohmic and fourier conduction. I have been recently thinking about this problem as it pertains to battery simulation. Quite an interesting modelling technique with broad similarities to PNMs for modelling batteries and electrochemical devices are equivalent circuit networks. These combine resistors and capacitors into unit cells which then become the building blocks of the model and a recent paper by a group at Imperial published one implementing an electrical and thermal equivalent circuit http://jes.ecsdl.org/content/165/13/A3169.short . 

The analogies between heat and electrical conduction are well known and summarized quite nicely in the intro of this paper: https://nvlpubs.nist.gov/nistpubs/jres/61/jresv61n2p105_A1b.pdf

It seems what we are actually solving in our analogous algorithms is an apparent diffusivity alpha and I think we need to make this distinction more clear. For heat transport alpha=k/(rho*C_p) where k is thermal conductivity, rho is density and C_p specific heat capacity. For Ohmic alpha= 1/(R*C) where C is capacitance per unit volume and 1/R is electrical conductivity but I think this only strictly applies when you have a shunt capacitor between every resistor in the network connected in parallel (as pictured in the above link). In some ways I guess the charging and discharging of a capacitor is similar to reactions and conversions of species.

So two things to discuss is 
1) How do we go about ensuring that the equations we are solving are physically representative. At the moment the onus is on the user to "handle" units but this has clearly led to confusion especially in the transient case. A possible way to address the issue would be to non-dimensionalize the transport algorithms and improve our higher level wrappers in order to make the units and  conversions simpler but this would probably require us to fully embrace the units package. 
2) Might it be worth looking at developing OpenPNM to be able to handle more complex electrical circuit configurations or as a start enable OpenPNM to output our "pure-resistor" networks to interface with ECN software such as simscape or xyce https://xyce.sandia.gov/ This may not be too hard to do in fact and could potentially allow us to leverage some impressive sounding parallel processing methods.

**Comment by jgostick** [comment]
Instead of using this proposed software, we'd be better off to double-down on the use of Petsc, who have recently added a [graph/network module](https://www.mcs.anl.gov/petsc/dmnetwork/index.html)
